### ISSUE TITLE
add UpdateVM setting to qubes-vm-settings

### ISSUE THREAD

**Issue by adrelanos** [QubesOS/qubes-issues#3412]
#### Qubes OS version:

R4

#### Affected TemplateVMs: 

dom0

---

### Steps to reproduce the behavior:
open `qubes-vm-settings`
### Expected behavior:
Should have UpdateVM setting.

### Actual behavior:
Does not have UpdateVM setting.

### General notes:

Cloning a TempalteVM is a natural step, even an encouraged one. The newly introduced burden in Qubes R4 that one has to edit dom0 configuration after cloning a template is a big usability issue.

NetVM being set to `none` for TemplateVMs (especially Whonix TemplateVMs) is very confusing for most users. (User question: "How shall I upgrade without a network connection.") Many users will think this is wrong and will manually set a NetVM.

Users who clone their `whonix-gw` / `whonix-ws` will very likely have a clearnet leak, will have them upgrades through clearnet rather than Tor which is clearly unexpected. (#3118) It is very non-obvious and difficult to learn that one has to edit dom0 `/etc/qubes-rpc/policy/qubes.UpdatesProxy`.

Therefore I suggest adding to `qubes-vm-settings` an UpdateVM setting below its NetVM setting.

I wouldn't call it UpdatesProxy rather than UpdateVM, because UpdateVM makes more sense for users.

---

#### Related issues:
#3118

**Comment by adrelanos** [comment]
Should this ticket have the `qubes-manager` tag?
Dunno if `qubes-vm-settings` is considered part of QVMM.
I think this is primarily a GUI rather than `core` issue.

**Comment by marmarta** [comment]
fixed and obsoleted by https://github.com/QubesOS/qubes-desktop-linux-manager/pull/133

**Comment by adrelanos** [comment]
Will there be an UpdateVM setting (to configure UpdatesProxy) in QVMM?

**Comment by marmarek** [comment]
yup, see the second screenshot at https://github.com/marmarta/qubes-config-manager#updates

**Comment by marmarta** [comment]
The thing is, there is no per-vm setting of "which update vm to use". There is only general policy. You can't really set it per-vm, not without meddling with multiple policy files.

**Comment by marmarek** [comment]
> Configurations such as "update Whonix Templates through sys-whonix but update all other Templates through sys-firewall" would still be difficult, require manual qrexec config file editing?

This is actually what this settings tab do: you have separate update proxy setting for Whonix (qubes with `whonix-updatevm` tag), and default for every other template. See the very central part of the screenshot you linked.

And in fact, configuration like this is a reason to _not_ put that into per-vm settings. Because then, you'd need to change it in several places (and likely miss some) to achieve this result.

> Another use case is cloning a Whonix Template. Which UpdateVM would be used for the cloned Whonix Template and how would the user see this, change this?

Tags are preserved during cloning, so it will still use the one for whonix.

**Comment by adrelanos** [comment]
Please re-open.

> > Configurations such as "update Whonix Templates through sys-whonix but update all other Templates through sys-firewall" would still be difficult, require manual qrexec config file editing?
> 
> This is actually what this settings tab do: you have separate update proxy setting for Whonix (qubes with `whonix-updatevm` tag), and default for every other template. See the very central part of the screenshot you linked.

I think users trying to debug issues similar to https://github.com/QubesOS/qubes-issues/issues/8490 are not looking there. Other things that users try is to create a clone of a Template (`whonix-gateway-clone`) and a new (or cloned) `sys-whonix-2` and want the `whonix-gateway-clone` updates through `sys-whonix-2`.

By using QVMM create/clone template and qubes-vm-settings it's not obvious at all that there's a separate UpdatesProxy setting. Even if users previously saw Qubes OS Global Config -> Updates, they might not have understood it or not remember about it in that moment.

Hence, I have an idea...

> And in fact, configuration like this is a reason to _not_ put that into per-vm settings. Because then, you'd need to change it in several places (and likely miss some) to achieve this result.

I see your point.

So how about an "informaion-only setting" in qubes-vm-settings?

In `qubes-vm-settings` an `UpdateVM` setting below its `NetVM` setting. But make it information only. Meaning, it will be grayed out. It cannot be changed there. And add an `I` (info) symbol right next to it. A tooltip would explain that this setting can be found Qubes OS Global Config -> Updates.
